src/job_description_feature_extraction.py

```diff
@@ -56,6 +60,58 @@ def get_feature_word_count(column_to_count):
     return count_dict
 
 
+def get_tfidf_similarity(corpus, queries):
+    """
+    calculate cosine similarity using tf.if between corpus an queries
+
+    :param corpus: list of text
+    :param queries: list of text
+    :return: matrix of ONLY corpus x query similarities
+    """
+
+    # # easier to test with smaller data set
+    # # use this to overwrite the incoming corpus/queries
+    # corpus = ['bob is in the house', 'susi goes to school', 'et tu brutu']
+    # queries = ['where is the school', 'bob is in the house']
+
+    tfidf_vectorizer = TfidfVectorizer()
+    tfidf_vectorizer.fit(corpus, queries)
+    corpus_vector = tfidf_vectorizer.transform(corpus)
+    queries_vector = tfidf_vectorizer.transform(queries)
+
+    similarity = cosine_similarity(queries_vector, corpus_vector)
+
+    # TODO: we can not use _vectorize here, because of the need of using the same vectorizer
+    # either rewrite it or remove it entirely, if get_top_idf_features is not going to be used
+    # corpus_len = len(corpus)
+    # queries_len = len(queries)
+
+    # we have to calculate the similarity for corpus + query to avoid dimension mismatch
+    # but we are only interested in the similarity of queries and documents
+    # remove all corpus x corpus similarities
+    # similarity = similarity[-queries_len:]
+    # remove all query x query similarities
+    # similarity = numpy.delete(similarity, range(corpus_len, corpus_len + queries_len), 1)
+
+    return similarity
+
+
+def extract_relevant_documents(similarity, sim_threshold=0.8):
+    """
+    return relevant documents from similarity matrix
+
+    :param similarity: similarity matrix from get_tfidf_similarity()
+    :param sim_threshold: cutoff thershold for relevant document (0.8) by default
+    :return: dict of (document_index, similarity) for every query from similarity matrix
+    """
+    relevant_documents = {}
+    for query_index, similarity_list in enumerate(similarity):
+        for document_index, similarity_score in enumerate(similarity_list):
+            if similarity_score > sim_threshold:
+                relevant_documents[query_index] = (document_index, similarity_score)
```

Comment:
this only keeps the 'last seen match' for any query index, should do something like append or pick the max.

```
ipdb> similarity
array([[ 0.51639778,  0.28867513,  0.        ],
       [ 1.        ,  0.        ,  0.        ]])
ipdb> extract_relevant_documents(similarity, sim_threshold=0.5)
{0: (0, 0.51639777949432231), 1: (0, 1.0000000000000002)}
ipdb> extract_relevant_documents(similarity, sim_threshold=0.2)
{0: (1, 0.28867513459481287), 1: (0, 1.0000000000000002)}
```

i.e: we may not get the best result.